src/azure-cli-core/azure/cli/core/commands/client_factory.py

```diff
@@ -194,6 +194,19 @@ def _get_mgmt_service_client(cli_ctx,
         client_kwargs.update(_prepare_client_kwargs_track2(cli_ctx))
         client_kwargs['credential_scopes'] = resource_to_scopes(resource)
 
+        # Track 2 currently lacks the ability to take external credentials.
+        #   https://github.com/Azure/azure-sdk-for-python/issues/8313
+        # As a temporary workaround, manually add external tokens to 'x-ms-authorization-auxiliary' header.
+        #   https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/authenticate-multi-tenant
+        if aux_subscriptions or aux_tenants:
+            _, _, _, external_tenant_tokens = cred.get_all_tokens(*resource_to_scopes(resource))
+
+            # external_tenant_tokens can be [] if no external tenant is involved.
+            if external_tenant_tokens:
+                # Hard-code scheme to 'Bearer' as _BearerTokenCredentialPolicyBase._update_headers does.
+                client_kwargs['headers']['x-ms-authorization-auxiliary'] = \
+                    ', '.join("Bearer {}".format(t[1]) for t in external_tenant_tokens)
```

Comment:
Can both `,` and `;` be used as the delimiter for each token?

Reply:
I simply don't know. We need to do some test. 

The exiting code 

https://github.com/Azure/azure-cli/blob/d07ee662c8cb49978901fc4dc11ddad46c89c27a/src/azure-cli-core/azure/cli/core/adal_authentication.py#L142

contradicts the document [Authenticate requests across tenants](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/authenticate-multi-tenant)

```
Bearer <auxiliary-token1>, EncryptedBearer <auxiliary-token2>, Bearer <auxiliary-token3>
```

That's why this PR is still marked as a **draft**. 😉

Reply:
According to my experiment of ARM deployment, the correct delimiter should be `, `.
When the delimiter is `;`, the service will return the following error:
```
{"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}}
```

Reply:
According to my experiment of ARM deployment, the correct delimiter should be `, `.
When the delimiter is `;`, the service returns the following error:
```
{"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}}
```

Reply:
True! Also verified with:

```
> az rest --url "https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2020-10-01" --headers "x-ms-authorization-auxiliary=Bearer token1; Bearer token2" --verbose
Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2020-10-01'
Request method: 'GET'
Request headers:
    'User-Agent': 'python/3.8.8 (Windows-10-10.0.19041-SP0) AZURECLI/2.20.0 (PIP)'
    'Accept-Encoding': 'gzip, deflate'
    'Accept': '*/*'
    'Connection': 'keep-alive'
    'x-ms-authorization-auxiliary': 'Bearer token1; Bearer token2'
    'x-ms-client-request-id': 'aad4a80f-5a67-425c-bc3d-9b5a2f471959'
    'CommandName': 'rest'
    'ParameterSetName': '--url --headers --verbose'
    'Authorization': 'Bearer eyJ0eXAiOiJKV...'
Request body:
None
Response status: 401
Response headers:
    'Cache-Control': 'no-cache'
    'Pragma': 'no-cache'
    'Content-Type': 'application/json; charset=utf-8'
    'Expires': '-1'
    'WWW-Authenticate': 'Bearer authorization_uri="https://login.windows.net/54826b22-38d6-4fb2-bad9-b7b93a3e9c5a", error="invalid_token", error_description="Invalid auxiliary tokens \'Bearer token1; Bearer token2\'."'
    'x-ms-failure-cause': 'gateway'
    'x-ms-request-id': '08deb134-9056-4afa-a468-9025f17246f3'
    'x-ms-correlation-request-id': '08deb134-9056-4afa-a468-9025f17246f3'
    'x-ms-routing-request-id': 'SOUTHEASTASIA:20210311T120202Z:08deb134-9056-4afa-a468-9025f17246f3'
    'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
    'X-Content-Type-Options': 'nosniff'
    'Date': 'Thu, 11 Mar 2021 12:02:01 GMT'
    'Connection': 'close'
    'Content-Length': '272'
Response content:
{"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}}
Unauthorized({"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}})
```

Reply:
True! Also verified with:

```
> az rest --url "https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2020-10-01" --headers "x-ms-authorization-auxiliary=Bearer token1; Bearer token2" --verbose
Request URL: 'https://management.azure.com/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourcegroups?api-version=2020-10-01'
Request method: 'GET'
Request headers:
    'Authorization': 'Bearer eyJ0eXAiOiJKV...'
    'x-ms-authorization-auxiliary': 'Bearer token1; Bearer token2'    
    ...

Response status: 401
Response headers:
    'WWW-Authenticate': 'Bearer authorization_uri="https://login.windows.net/54826b22-38d6-4fb2-bad9-b7b93a3e9c5a", error="invalid_token", error_description="Invalid auxiliary tokens \'Bearer token1; Bearer token2\'."'
    ...
Response content:
{"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}}
Unauthorized({"error":{"code":"InvalidAuxiliaryTokens","message":"Authentication failed for auxiliary token: 'The auxiliary tokens should have proper format \"'scheme' 'value'\". The '1' auxiliary token(s) were not found in proper format. Invalid tokens are in the response header.'"}})
```

This means the doc is right and the current CLI code is wrong.